src/jit/compiler.h

```diff
@@ -7984,7 +7984,9 @@ public :
         // Methods returning a struct in two registers is considered having a return value of TYP_STRUCT.
         // Such method's compRetNativeType is TYP_STRUCT without a hidden RetBufArg
         return varTypeIsStruct(info.compRetNativeType) && (info.compRetBuffArg == BAD_VAR_NUM);
-#else 
+#elif FEATURE_MULTIREG_RET && defined(_TARGET_X86_)
+        return varTypeIsLong(info.compRetNativeType);
```

Comment:
I see this method is called by fgAddInternal() to generate a temp for single block return value.  That code is making the assumption that if compMethodReturnsMultiRegRetType() is true then it is TYP_STRUCT.  That would work for unix and Arm64 but not for x86/arm32.

I think we need to fix the following line in fgAddInternal() as follows (high-lighted in bold)

        _else if (compMethodReturnsMultiRegRetType())
        {
            **lvaTable[genReturnLocal].lvType = genActualType(info.compRetNativeType)**
            lvaSetStruct(genReturnLocal, info.compMethodInfo->args.retTypeClass, true);
            lvaTable[genReturnLocal].lvIsMultiRegArgOrRet = true;
        }_